### PR TITLE
JDK-8303349: Simplify link format for generic types in index pages

### PR DESCRIPTION
**Pull request by hns**
Please review a simple change to simplify the format of links to generic types in various JavaDoc index pages. Previously, these included dedicated links for type parameters and bounds, which seems excessive for use in an index. 

Below are before and after screenshots for `Spliterator.ofPrimitive` (which is admittedly an extreme case) and neighboring index entries. Note that now links to generic types match what we do for executables, where the whole signature is used as link label.

Before:
index-generic-links-1

After:
index-generic-links-2

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8303349](https://bugs.openjdk.org/browse/JDK-8303349): Simplify link format for generic types in index pages

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to [eeb0540d](https://git.openjdk.org/jdk/pull/12786/files/eeb0540d11c26111afa816962a6251e1502a7825)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12786/head:pull/12786` \
`$ git checkout pull/12786`

Update a local copy of the PR: \
`$ git checkout pull/12786` \
`$ git pull https://git.openjdk.org/jdk pull/12786/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12786`

View PR using the GUI difftool: \
`$ git pr show -t 12786`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12786.diff">https://git.openjdk.org/jdk/pull/12786.diff</a>

</details>

### CODE REVIEW

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlLinkInfo.java [hunk 1]
```diff
@@ -45,27 +45,33 @@ public class HtmlLinkInfo extends LinkInfo {
 
     public enum Kind {
         /**
-         * Link with just the element name as label.
+         * Creates a link with just the element name as label.
```
Comment (jonathan-gibbons) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlLinkInfo.java:48]: (Minor)
The use of `Creates` makes this seem like a doc comment for a method. Enum constants are state/variables, not methods.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlLinkInfo.java [hunk 2]
```diff
@@ -228,7 +234,10 @@ public boolean isLinkable() {
 
     @Override
     public boolean showTypeParameters() {
-        return context != Kind.PLAIN && context != Kind.SHOW_PREVIEW;
+        // Type parameters parameters for these kinds of links are either not desired
```
Comment (jonathan-gibbons) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlLinkInfo.java:237]: duplicate word `parameters`

**Review by jonathan-gibbons — APPROVED** [review]
Approved, with quibbles on comments for your consideration.

**Comment by hns** [comment]
Thanks for the review, I addressed the feedback.

**Pull request closed without merging**